compute_sdk/globus_compute_sdk/sdk/executor.py

```diff
@@ -1274,6 +1274,7 @@ def _match_results_to_futures(self):
         for fut in futures_to_complete:
             props, res = self._received_results.pop(fut.task_id)
 
+            self.funcx_executor.client._log_version_mismatch(res.details)
             if res.is_error:
```

Comment:
Slightly weird to be accessing the (nominally) private member.  I wonder if we can be creative with the method name or interface so that this is cleaner.  Clearly, there's an external-to-Client need to access this method's logic, so...